lib/src/common/directives/ng_style.dart

```diff
@@ -11,58 +11,39 @@ import "package:angular2/src/facade/lang.dart" show isPresent, isBlank;
 import "../../core/change_detection/differs/default_keyvalue_differ.dart"
     show KeyValueChangeRecord;
 
-/// The `NgStyle` directive changes styles based on a result of expression
-/// evaluation.
+/// The `NgStyle` directive changes an element's style based on the bound style
+/// expression:
 ///
-/// An expression assigned to the `ngStyle` property must evaluate to an object
-/// and the corresponding element styles are updated based on changes to this
-/// object. Style names to update are taken from the object's keys, and values
-/// - from the corresponding object's values.
+///     <div [ngStyle]="styleExp"></div>
 ///
-/// ### Syntax
+/// _styleExp_ must evaluate to a `Map<String, String>`. Element style properties
+/// are set based on the map entries: each _key_:_value_ pair identifies a
+/// style property _name_ and its _value_.
 ///
-/// - `<div [ngStyle]="{'font-style': style}"></div>`
-/// - `<div [ngStyle]="styleExp"></div>` - here the `styleExp` must evaluate to
-/// an object
+/// See the [Template Syntax section on `NgStyle`][guide] for more details.
 ///
-/// ### Example:
+/// ### Examples
 ///
-/// ```dart
-/// import 'angular2/core.dart' Component;
-/// import 'angular2/common.dart' NgStyle;
+/// Try the [live example][ex] from the [Template Syntax][guide] page. Here are
+/// the relevant excerpts from the example's template and the corresponding
+/// component class:
 ///
-/// @Component(
-///  selector: 'ngStyle-example',
-///  template: '''
-///    <h1 [ngStyle]="{'font-style': style, 'font-size': size, 'font-weight': weight}">
-///      Change style of this text!
-///    </h1>
+/// {@example docs/template-syntax/lib/app_component.html region=NgStyle}
 ///
-///    <hr>
+/// {@example docs/template-syntax/lib/app_component.dart region=NgStyle}
```

Comment:
Yes, that is referring to a code excerpt from the angular.io template-syntax example.

It all gets pulled together when angular.io is built.

The folder name in the `@example` URI is "docs" because that is what is used in the angular.io URL; e.g., Dart docs are rooted at https://angular.io/docs/dart/latest/index.html